gnomad/resources/resource_utils.py

```diff
@@ -351,6 +368,59 @@ def __init__(self, default_version: str, versions: Dict[str, BlockMatrixResource
         super().__init__(default_version, versions)
 
 
+GNOMAD_PUBLIC_BUCKETS = ("gnomad-public", "gnomad-public-requester-pays")
+
+
+class GnomadPublicResource(BaseResource, ABC):
+    """Base class for the gnomAD project's public resources."""
+
+    def _get_path(self) -> str:
+        resource_source = gnomad_public_resource_configuration.source
+        if resource_source == GnomadPublicResourceSource.GNOMAD:
+            return self._path
+
+        relative_path = reduce(
+            lambda path, bucket: path[5 + len(bucket) :]
+            if path.startswith(f"gs://{bucket}")
+            else path,
+            GNOMAD_PUBLIC_BUCKETS,
+            self._path,
+        )
+
+        if resource_source == GnomadPublicResourceSource.GOOGLE_CLOUD_PUBLIC_DATASETS:
+            return f"gs://gcp-public-data--gnomad{relative_path}"
+
+        return (
+            f"{resource_source.rstrip('/')}{relative_path}"  # pylint: disable=no-member
```

Comment:
Couldn't figure out why Pylint insisted this was an error. The tests show that this works.

Pylint also only complained about this when running in the PR checks. It passed when running Pylint locally.